packages/edit-site/src/components/list/customized-template-info.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+export default function CustomizedTemplateInfo( { template } ) {
+	// Template originally provided by a theme, but customized by a user.
+	// Templates originally didn't have the 'origin' field so identify
+	// older customized templates by checking for no origin and a 'theme'
+	// or 'custom' source.
+	if (
+		template.author &&
+		template.has_theme_file &&
+		( template.origin === 'theme' ||
+			( ! template.origin &&
+				[ 'theme', 'custom' ].includes( template.source ) ) ||
+			template.origin === 'plugin' )
```

Comment:
It's pretty complicated because this combines all the logic into one, and it might be clearer to extract it into separate variables:
```js
// Templates originally didn't have the 'origin' field so identify
// older customized templates by checking for no origin and a 'theme'
// or 'custom' source.
const isAddedByTheme = template.has_theme_file &&
			( template.origin === 'theme' ||
				( ! template.origin &&
					[ 'theme', 'custom' ].includes( template.source ) ) );
const isAddedByPlugin = template.has_theme_file && template.origin === 'plugin';
const isCustomized = template.source === 'custom';

if ( ( isAddedByTheme || isAddedByPlugin ) && isCustomized ) {
```